katello/katello-backup

```diff
@@ -63,59 +100,132 @@ def create_directories(directory)
   FileUtils.chmod_R 0770, directory
 end
 
-def backup_db_online
-  puts "Backing up postgres online schema... "
-  run_cmd("runuser - postgres -c 'pg_dumpall -g > #{@dir}/pg_globals.dump'")
-  run_cmd("runuser - postgres -c 'pg_dump -Fc foreman > #{@dir}/foreman.dump'")
-  run_cmd("runuser - postgres -c 'pg_dump -Fc candlepin > #{@dir}/candlepin.dump'")
-  puts "Done."
+def snapshot_backup
+  @mountdir = @options[:snapshot_mount] || "/var/snap"
+  FileUtils.mkdir_p @mountdir
+  run_cmd("katello-service stop #{EXCLUDED}")
+  create_and_mount_snapshots
+  run_cmd("katello-service start #{EXCLUDED}")
+  backup_and_destroy_snapshots
+end
 
-  puts "Backing up mongo online schema... "
-  run_cmd('mongodump --host localhost --out mongo_dump')
-  puts "Done."
+def create_and_mount_snapshots
+  @databases.each do |database|
```

Comment:
this creates one snapshot per database, in the case where multiple dbs are on the same LV, we thus create multiple snaps of the same data. not too bad (we do that in the "split everything" case too) but one could safe a bit of space here. would make the code more complicated, though